dandi/dandiarchive.py

```diff
@@ -22,8 +16,200 @@
 lgr = get_logger()
 
 
+class ParsedDandiURL(ABC, BaseModel):
```

Comment:
Please add high level description for this base class and what to expect subclasses to override (seems to be `get_assets`?) and may be that all methods but `get_client` and `navigate` expect client to be passed ?

Reply:
Done.